plugins/auto-sizes/hooks.php

```diff
@@ -15,10 +15,14 @@
  *
  * @since 1.0.0
  *
- * @param array $attr Attributes for the image markup.
- * @return array The filtered attributes for the image markup.
+ * @param array<string, string>|mixed $attr Attributes for the image markup.
```

Comment:
This is always expected to be an array. Would it be better to add strong typing to the param, or are you trying to avoid doing so because it is filtered content? I have similar questions about several other places where similar changes are being applied.

Reply:
Yes, because it is filtered another plugin would put anything here. So it seems safer to use `mixed` and then do runtime type checking to avoid a fatal error.

Reply:
Would you want a `_doing_it_wrong()` to be added in cases like this so that the error isn't silently suppressed?

Reply:
If the strategy is to treat all input from filters as `$mixed` then I think this is fine for consistency. Making sure that we're doing early type validation—as you're doing—seems like enough to me.